src/scripts/main.js

```diff
@@ -1,3 +1,59 @@
 'use strict';
 
-// write code here
+const table = document.querySelector('table.field');
+const buttons = document.querySelectorAll('button.button');
+
+buttons.forEach((button) => {
+  button.addEventListener('click', () => {
+    const tbody = table.querySelector('tbody') || table;
+
+    if (button.classList.contains('append-row')) {
+      if (table.rows.length >= 10) {
+        return;
+      }
+
+      const newRow = table.insertRow();
+
+      for (let i = 0; i < table.rows[0].cells.length; i++) {
+        newRow.insertCell();
+      }
+
+      document.querySelector('.remove-row').disabled = false;
+
+      if (table.rows.length >= 10) {
+        button.disabled = true;
+      }
+    } else if (button.classList.contains('remove-row')) {
+      tbody.deleteRow(-1);
+      document.querySelector('.append-row').disabled = false;
+
+      if (table.rows.length <= 2) {
+        button.disabled = true;
+      }
+    } else if (button.classList.contains('append-column')) {
+      if (table.rows[0].cells.length >= 10) {
```

Comment:
The condition `table.rows[0].cells.length >= 10` is used to prevent adding more than 10 columns. Ensure that this limit aligns with your requirements. If the maximum number of columns should be different, adjust this value accordingly.